Blizzard.IO.RabbitMQ.Builders/BaseNetqRabbitBuilder.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections.Generic;
+using EasyNetQ;
+using Blizzard.IO.RabbitMQ.Extensions;
+
+namespace Blizzard.IO.RabbitMQ.Builders
+{
+    public abstract class BaseNetqRabbitBuilder
+    {
+        private static Dictionary<string, IBus> netqRabbitConnections = new Dictionary<string, IBus>();
+
+        protected IBus InitConnection(
+            string host,
+            string password,
+            string username,
+            ushort timeout = 10,
+            string product = "",
+            string platform = "",
+            string virtualHost = "/",
+            int requestHeartbeat = 10,
+            int prefetchCount = 50,
+            bool publisherConfirms = false,
+            bool persistentMessages = true)
+        {
+            string busKey = $"{host}{username}{password}";
```

Comment:
What you think about making it a struct or class with equal ovverride (Name it ConnectionKey or something like that) ? It doesn't look very good like this.

Reply:
It will be more problematic to check if some `ConnectionKey` already exists, since a reference type object will now be the `TKey` type in the dict

Reply:
כ

Reply:
thats why I did a struct which is not a reference typr